x-pack/plugins/infra/public/pages/logs/log_entry_categories/page_providers.tsx

```diff
@@ -7,30 +7,46 @@
 
 import React from 'react';
 import { LogAnalysisSetupFlyoutStateProvider } from '../../../components/logging/log_analysis_setup/setup_flyout';
+import { LogSourceErrorPage } from '../../../components/logging/log_source_error_page';
+import { SourceLoadingPage } from '../../../components/source_loading_page';
 import { LogEntryCategoriesModuleProvider } from '../../../containers/logs/log_analysis/modules/log_entry_categories';
 import { useLogSourceContext } from '../../../containers/logs/log_source';
 import { useActiveKibanaSpace } from '../../../hooks/use_kibana_space';
 
 export const LogEntryCategoriesPageProviders: React.FunctionComponent = ({ children }) => {
-  const { sourceId, resolvedSourceConfiguration } = useLogSourceContext();
+  const {
+    hasFailedLoading,
+    isLoading,
+    isUninitialized,
+    latestLoadSourceFailures,
+    loadSource,
+    resolvedSourceConfiguration,
+    sourceId,
+  } = useLogSourceContext();
   const { space } = useActiveKibanaSpace();
 
   // This is a rather crude way of guarding the dependent providers against
   // arguments that are only made available asynchronously. Ideally, we'd use
   // React concurrent mode and Suspense in order to handle that more gracefully.
-  if (!resolvedSourceConfiguration || space == null) {
+  if (space == null) {
+    return null;
+  } else if (hasFailedLoading) {
```

Comment:
The guards had to move up a level in the hierarchy, cause the loading screen and errors wouldn't be rendered otherwise.